statesync/reactor_test.go

```diff
@@ -284,3 +478,87 @@ func retryUntil(t *testing.T, fn func() bool, timeout time.Duration) {
 		require.NoError(t, ctx.Err())
 	}
 }
+
+func handleLightBlockRequests(t *testing.T,
+	chain map[int64]*types.LightBlock,
+	receiving chan p2p.Envelope,
+	sending chan p2p.Envelope,
+	close chan struct{},
+	failureRate int) {
+	requests := 0
+	for {
+		select {
+		case envelope := <-receiving:
+			if msg, ok := envelope.Message.(*ssproto.LightBlockRequest); ok {
+				if requests%10 >= failureRate {
+					lb, err := chain[int64(msg.Height)].ToProto()
+					require.NoError(t, err)
+					sending <- p2p.Envelope{
+						From: envelope.To,
+						Message: &ssproto.LightBlockResponse{
+							LightBlock: lb,
+						},
+					}
+				} else {
+					switch rand.Intn(3) {
+					case 0: // send a different block
+						differntLB, err := mockLB(t, int64(msg.Height), factory.DefaultTestTime, factory.MakeBlockID()).ToProto()
+						require.NoError(t, err)
+						sending <- p2p.Envelope{
+							From: envelope.To,
+							Message: &ssproto.LightBlockResponse{
+								LightBlock: differntLB,
+							},
+						}
+					case 1: // send nil block i.e. pretend we don't have it
+						sending <- p2p.Envelope{
+							From: envelope.To,
+							Message: &ssproto.LightBlockResponse{
+								LightBlock: nil,
+							},
+						}
+					case 2: // don't do anything
+					}
+				}
+			}
+		case <-close:
+			return
+		}
+		requests++
```

Comment:
do we use requests?

Reply:
Yeah we keep track of them to decide whether to respond with a correct block or a faulty one. See:
```golang
if requests%10 >= failureRate {
```